wz_strategic_plugin/src/wz_strategic_plugin.cpp

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "wz_strategic_plugin/wz_strategic_plugin.h"
+
+namespace wz_strategic_plugin
+{
+
+    void WzStrategicPlugin::initialize()
+    {
+        nh_.reset(new ros::CARMANodeHandle());
+        pnh_.reset(new ros::CARMANodeHandle("~"));
+        plan_maneuver_srv_ = nh_->advertiseService("plugins/RouteFollowing/plan_maneuvers", &WzStrategicPlugin::planManeuverCb, this);
+
+        plugin_discovery_pub_ = nh_->advertise<cav_msgs::Plugin>("plugin_discovery", 1);
+        plugin_discovery_msg_.name = "WzStrategic";
+        plugin_discovery_msg_.versionId = "v1.0";
+        plugin_discovery_msg_.available = true;
+        plugin_discovery_msg_.activated = true;
+        plugin_discovery_msg_.type = cav_msgs::Plugin::STRATEGIC;
+        plugin_discovery_msg_.capability = "strategic_plan/plan_maneuvers";
+
+        pose_sub_ = nh_->subscribe("current_pose", 1, &WzStrategicPlugin::pose_cb, this);
+        twist_sub_ = nh_->subscribe("current_velocity", 1, &WzStrategicPlugin::twist_cb, this);
+
+        wml_.reset(new carma_wm::WMListener());
+        // set world model point form wm listener
+        wm_ = wml_->getWorldModel();
+
+        discovery_pub_timer_ = pnh_->createTimer(
+            ros::Duration(ros::Rate(10.0)),
+            [this](const auto &) { plugin_discovery_pub_.publish(plugin_discovery_msg_); });
+
+    }
+
+    void WzStrategicPlugin::pose_cb(const geometry_msgs::PoseStampedConstPtr &msg)
+    {
+        pose_msg_ = msg;
+        lanelet::BasicPoint2d curr_loc(pose_msg_->pose.position.x, pose_msg_->pose.position.y);
+        current_loc_ = curr_loc;
+    }
+
+    void WzStrategicPlugin::twist_cb(const geometry_msgs::TwistStampedConstPtr &msg)
+    {
+        current_speed_ = msg->twist.linear.x;
+    }
+
+    std::string WzStrategicPlugin::traffic_light_interpreter(lanelet::CarmaTrafficLightState& state)
```

Comment:
Looks like this method is returning the same string for each state. Can you add a To Do comment if this method needs to be revised later?

Reply:
Also can we add a default case to the switch statement

Reply:
default case does not exist in this condition.